reactor-netty5-http/src/main/java/reactor/netty5/http/server/HttpServerFormDecoderProvider.java

```diff
@@ -296,4 +335,120 @@ HttpServerFormDecoderProvider build() {
 			return new HttpServerFormDecoderProvider(this);
 		}
 	}
+
+	static final class ReactorNettyHttpPostMultipartRequestDecoder extends HttpPostMultipartRequestDecoder
+			implements ReactorNettyHttpPostRequestDecoder {
+
+		/**
+		 * Current {@link HttpData} from the body (only the completed {@link HttpData}).
+		 */
+		final List<HttpData> currentCompletedHttpData = new ArrayList<>();
+
+		ReactorNettyHttpPostMultipartRequestDecoder(HttpDataFactory factory, HttpRequest request) {
+			super(factory, request);
+		}
+
+		@Override
+		protected void addHttpData(InterfaceHttpData data) {
+			if (data instanceof HttpData httpData) {
+				currentCompletedHttpData.add(httpData);
+			}
+		}
+
+		@Override
+		public void cleanCurrentHttpData(boolean onlyCompleted) {
+			for (HttpData data : currentCompletedHttpData) {
+				removeHttpDataFromClean(data);
+				data.close();
+			}
+			currentCompletedHttpData.clear();
+
+			if (!onlyCompleted) {
+				InterfaceHttpData partial = currentPartialHttpData();
+				if (partial instanceof HttpData partialData) {
+					partialData.delete();
+				}
+			}
+		}
+
+		@Override
+		public List<HttpData> currentHttpData(boolean onlyCompleted) {
+			if (!onlyCompleted) {
+				InterfaceHttpData partial = currentPartialHttpData();
+				if (partial instanceof HttpData partialData) {
+					partialData.usingContent(partialContent ->
```

Comment:
Which operation is throwing an exception so that we need to use lambda?

Reply:
since the _usingContent_ method is not yet finalized in the codec-multipart, I have committed a conservative and safe patch in [868e1bd](https://github.com/reactor/reactor-netty/pull/2327/commits/868e1bdd9cbdd5fe91e4c2896d70d3b055978220), which copies the partial content into the currentCompletedHttpData list. 

this is not optimal, but once the multiplart codec will be finalized, we can then revisit the way partial data are handled.